lib/vector/Vlib/map.c

```diff
@@ -384,7 +384,7 @@ int Vect_delete(const char *map)
 int Vect__delete(const char *map, int is_tmp)
 {
     int ret;
-    char path[GPATH_MAX], path_buf[GPATH_MAX];
+    char path[GPATH_MAX], path_buf[GPATH_MAX + GPATH_MAX];
```

Comment:
`GPATH_MAX` must not be exceeded, thus please leave it at `GPATH_MAX` and add a check that `path, ent->d_name` do not exceed `GPATH_MAX - 2`

Reply:
I updated with a suggestion using `snprintf()`, which is C99 feat, but perhaps give a safer and cleaner code.

Reply:
We have `G_snprintf()`

Reply:
Interestingly  `G_snprintf` calls `vsnprintf`, which is C99 feat.

Reply:
In this case I prefer `snprintf` over `G_snprintf`. @neteler you being the author of `G_snprintf`, can you remember the reason for the existence of `G_snprintf`?

Reply:
Honestly, I don't. But this is the original commit: https://github.com/OSGeo/grass-legacy/commit/e8361476c09903d60f29dbda8e1d7b0fd43d443f#diff-87363de8ff12b50f8adbfd1fb5baefb5ef308b1fc029a16be86c6e688dbf6beb

which contains a "use is discouraged" comment ...

Reply:
Comparing the implementation of G_snprintf to glibc and Darwin/FreeBSD, there is one difference that stands out: a fix for Windows. This limitation/bug with Visual Studio has been fixed since its 2015 version at the latest (according to my investigations). I'm actually planning to propose a PR later on to update/drop G_snprintf.

Reply:
From deep in the archives: https://lists.osgeo.org/pipermail/grass-dev/2002-April/005142.html